lib/datasource/go/index.ts

```diff
@@ -172,11 +196,23 @@ export async function getDigest(
   value?: string
 ): Promise<string | null> {
   const source = await getDatasource(lookupName);
-  if (source && source.datasource === github.id) {
-    // ignore v0.0.0- pseudo versions that are used Go Modules - look up default branch instead
-    const tag = value && !value.startsWith('v0.0.0-2') ? value : undefined;
-    const digest = await github.getDigest(source, tag);
-    return digest;
+  if (!source) {
+    return null;
+  }
+
+  // ignore v0.0.0- pseudo versions that are used Go Modules - look up default branch instead
+  const tag = value && !value.startsWith('v0.0.0-2') ? value : undefined;
+
+  switch (source.datasource) {
+    case github.id: {
+      return github.getDigest(source, tag);
+    }
+    case bitbucket.id: {
+      return bitbucket.getDigest(source, tag);
+    }
+    /* istanbul ignore next */
+    default: {
+      return null;
+    }
```

Comment:
Please add test if possible

Reply:
Is this nor testable too?

Reply:
It is possible to get here when gitlab is used as datastore (since `getDigest` is not implemented for gitlab).
I added tests for that case.

Reply:
See here:
https://github.com/renovatebot/renovate/pull/7892/files#diff-9bf3149428b9f2fc82e6409c493486601571a66b462b8eac2bd6c98e66d8c94bR60-R71